avm-transpiler/src/transpile.rs

```diff
@@ -0,0 +1,289 @@
+                    BinaryFieldOp::Add => AvmOpcode::ADD,
+                    BinaryFieldOp::Sub => AvmOpcode::SUB,
+                    BinaryFieldOp::Mul => AvmOpcode::MUL,
+                    BinaryFieldOp::Div => AvmOpcode::DIV,
+                    BinaryFieldOp::Equals => AvmOpcode::EQ,
+                };
+                // TODO(4268): set in_tag to `field`
+                avm_instrs.push(AvmInstruction {
+                    opcode: avm_opcode,
+                    operands: vec![
+                        AvmOperand::U32 {
+                            value: lhs.to_usize() as u32,
+                        },
+                        AvmOperand::U32 {
+                            value: rhs.to_usize() as u32,
+                        },
+                        AvmOperand::U32 {
+                            value: destination.to_usize() as u32,
+                        },
+                    ],
+                    ..Default::default()
+                });
+            }
+            BrilligOpcode::BinaryIntOp {
+                destination,
+                op,
+                bit_size: _, // TODO(4268): support u8..u128 and use in_tag
+                lhs,
+                rhs,
+            } => {
+                let avm_opcode = match op {
+                    BinaryIntOp::Add => AvmOpcode::ADD,
+                    BinaryIntOp::Sub => AvmOpcode::SUB,
+                    BinaryIntOp::Mul => AvmOpcode::MUL,
+                    BinaryIntOp::UnsignedDiv => AvmOpcode::DIV,
+                    BinaryIntOp::Equals => AvmOpcode::EQ,
+                    BinaryIntOp::LessThan => AvmOpcode::LT,
+                    BinaryIntOp::LessThanEquals => AvmOpcode::LTE,
+                    BinaryIntOp::And => AvmOpcode::AND,
+                    BinaryIntOp::Or => AvmOpcode::OR,
+                    BinaryIntOp::Xor => AvmOpcode::XOR,
+                    BinaryIntOp::Shl => AvmOpcode::SHL,
+                    BinaryIntOp::Shr => AvmOpcode::SHR,
+                    _ => panic!(
+                        "Transpiler doesn't know how to process BinaryIntOp {:?}",
+                        brillig_instr
+                    ),
+                };
+                // TODO(4268): support u8..u128 and use in_tag
+                avm_instrs.push(AvmInstruction {
+                    opcode: avm_opcode,
+                    operands: vec![
+                        AvmOperand::U32 {
+                            value: lhs.to_usize() as u32,
+                        },
+                        AvmOperand::U32 {
+                            value: rhs.to_usize() as u32,
+                        },
+                        AvmOperand::U32 {
+                            value: destination.to_usize() as u32,
+                        },
+                    ],
+                    ..Default::default()
+                });
+            }
+            BrilligOpcode::CalldataCopy { destination_address, size, offset } => {
```

Comment:
wait brillig already had calldatacopy, the more you know! how is this activated?

Reply:
Alvaro just added it! It's now there for all Brillig functions.